src/uct/ib/rc/base/rc_iface.c

```diff
@@ -260,16 +283,22 @@ ucs_status_t uct_rc_iface_handle_fc(uct_rc_iface_t *iface, unsigned qp_num,
 
     } else if (fc_hdr & UCT_RC_EP_FC_FLAG_HARD_REQ) {
         UCS_STATS_UPDATE_COUNTER(ep->fc.stats, UCT_RC_FC_STAT_RX_HARD_REQ, 1);
+        fc_req = ucs_mpool_get(&iface->tx.fc_mp);
+        if (ucs_unlikely(fc_req == NULL)) {
+            ucs_fatal("Failed to allocated FC request.");
```

Comment:
It can, but if we do not send a reply the communication will just stop. Sender will never get grant.